### ISSUE TITLE
Today's build 3.5.0+654 is not working properly for me

### ISSUE THREAD

**Issue by RawConvert** [darktable-org/darktable#7875]
Ubuntu 18.04    dt 3.5.0 Master

I think something basic has gone bad with dt or my system having built today and then tried it.

1. It won't apply my styles which were working fine in 3.4.  It says it's applied it, but it hasn't and the history confirms that.  Here's a style that doesn't work (I have two, neither work). Rename to .dtstyle of course and import.
[Style20.txt](https://github.com/darktable-org/darktable/files/5823428/Style20.txt)

2. Clicking an image in lighttable - it's not being selected.

3. The Import options in preferences have disappeared.

**Comment by Nilvus** [comment]
Probably your system. Import tab is still there in preferences on master in my side and clicking an image in lighttable is ok too. I rarely use styles to answer to first one. Do you always remove build folder in source code folder and destination install one of darktable? It's highly recommended to remove them to avoid some strange issues.

**Comment by RawConvert** [comment]
I still have the Import tab but within it, only Session Options remain, the rest (import options) is absent.

**Comment by RawConvert** [comment]
Re. removing folders - no, I wasn't aware of this, thanks.  Let me check my understanding...
1. Within my overall building folder I have "darktable-code" and within that "build".  I think you're saying that as soon as I've built my new dt, I should delete "build".  Is that right?
2. Re. "destination install one of darktable"  I don't understand, please could you explain what I need to do.

**Comment by Nilvus** [comment]
> I still have the Import tab but within it, only Session Options remain, the rest (import options) is absent.

That is normal. The rest have been moved in import modules. We will now add in lighttable, some module settings directly in "preset" menu. For import module, you will see a new parameters expander. That allow to change more quickly as wanted (and eventually for each import) those parameters.

> * Within my overall building folder I have "darktable-code" and within that "build".  I think you're saying that as soon as I've built my new dt, I should delete "build".  Is that right?

Yes, that's this one that you should remove before each new compile.

> * Re. "destination install one of darktable"  I don't understand, please could you explain what I need to do.

That's the folder where darktable will be installed. For example: /opt/darktable. It's better to remove it also before compile.

To compile quickly, I use the script below (it also guarantee that submodule, like rawspeed for example are updated):

```
### Building manually

#!/bin/sh

cd /home/nilvus/Projets/darktable
rm -rf build /opt/darktable
git pull#!/bin/sh

cd /home/nilvus/Projets/darktable
rm -rf build /opt/darktable
git pull
git submodule update
./build.sh --prefix /opt/darktable --build-type Release --install

exit 0
git submodule update
./build.sh --prefix /opt/darktable --build-type Release --install
```
exit 0
```

Just update paths of course regarding your folders. If all this fix your issues, think about closing this one.

**Comment by chhil** [comment]
![image](https://user-images.githubusercontent.com/1119439/104812081-d4abdf80-5825-11eb-94a4-ab61fd73f08a.png)

**Comment by Nilvus** [comment]
_A big mistake in my same comment first posted, so please ignore it if you already receive first one by email. Just consider this one instead:_

> I still have the Import tab but within it, only Session Options remain, the rest (import options) is absent.

That is normal. The rest have been moved in import modules. We will now add in lighttable, some module settings directly in "preset" menu. For import module, you will see a new parameters expander. That allow to change more quickly as wanted (and eventually for each import) those parameters.

> * Within my overall building folder I have "darktable-code" and within that "build".  I think you're saying that as soon as I've built my new dt, I should delete "build".  Is that right?

Yes, that's this one that you should remove before each new compile.

> * Re. "destination install one of darktable"  I don't understand, please could you explain what I need to do.

That's the folder where darktable will be installed. For example: /opt/darktable. It's better to remove it also before compile.

To compile quickly, I use the script below (it also guarantee that submodule, like rawspeed for example are updated):

```
### Building manually

#!/bin/sh

cd /home/nilvus/Projets/darktable
rm -rf build /opt/darktable### Building manually

#!/bin/sh

cd /home/nilvus/Projets/darktable
rm -rf build /opt/darktable
git pull#!/bin/sh

cd /home/nilvus/Projets/darktable
rm -rf build /opt/darktable
git pull
git submodule update
./build.sh --prefix /opt/darktable --build-type Release --install

exit 0
git pull
git submodule update
./build.sh --prefix /opt/darktable --build-type Release --install

exit 0
```

Just update paths of course regarding your folders. If all this fix your issues, think about closing this one.

**Comment by RawConvert** [comment]
Thanks for the tips.  I've just rebuilt Master, 3.5.0+753~gf8dce16ba-dirty, hoping it would now be ok, but it's the same, selection by mouse is not working.
Select All - works, the border area around each image changes.
Select None - works...
Hover over an image - it lights up but not in all cases by any means.  It's as if there is a mouse problem. The image hovered over does not light up.  You move on to a different one, it lights up.
Click an unselected image - it does not change
Click a selected image - it does not change
Double click image - goes into darkroom as expected.
All this is with elegant grey theme but it's the same with "darktable" theme.
Hover mouse from neutral area to an image and an orange border appears around the image AND an adjacent one, why a second one?  See screen grab.
I'm not aware of any issues with my mouse when using my system generally.
When you return to lighttable from dark, a thin border around the image slowly pulsates a few times, guess this is someone's feature in progress?
![Screenshot from 2021-01-19 18-10-35](https://user-images.githubusercontent.com/16868433/105076068-82b3c580-5a82-11eb-952d-0eb6f4e2db68.png)

Here is the log of the build in case it helps.
[DT build log 19Jan2021.txt](https://github.com/darktable-org/darktable/files/5837792/DT.build.log.19Jan2021.txt)

@Nilvus , I've been cloning into a brand new parent folder each time, e.g. today "DT-Mast-19Jan20" under Home.  So there are no leftover files to mess things up, if I understand correctly, re. your advice above about removing certain folders.

**Comment by RawConvert** [comment]
Ah, just realised the orange border is for when raw+jpeg, I don't normally import jpegs but hadn't got as far as setting that  option...

**Comment by Nilvus** [comment]
> Hover over an image - it lights up but not in all cases by any means. It's as if there is a mouse problem. The image hovered over does not light up. You move on to a different one, it lights up.

If it's already hovered, it's normal that the image is not hovered more. It's already the case (or if it's not that, it's not clear here).

> Click an unselected image - it does not change
> Click a selected image - it does not change

Normal behavior, you select or unselect an image by clicking on another (or button you describe in module in right panel). No issue here.

> All this is with elegant grey theme but it's the same with "darktable" theme.

It's not CSS related, so should work the same in all themes.

> I'm not aware of any issues with my mouse when using my system generally.

And on what you describe, I see no issue at all.

> When you return to lighttable from dark, a thin border around the image slowly pulsates a few times, guess this is someone's feature in progress?

It's not a feature in progress, but a sort of "flash" to quickly show selected images when coming back. And it's not new, as it is there since last spring.

> @Nilvus , I've been cloning into a brand new parent folder each time, e.g. today "DT-Mast-19Jan20" under Home. So there are no leftover files to mess things up, if I understand correctly, re. your advice above about removing certain folders.

Yes, install folder AND build folder in source code (where you git clone darktable repository) needs to be removed to be sure you have a clean darktable and so what developers have made.

To resume all that: I see no more issues remaining. The only issues you describe are just normal behavior. I'll close that one as we also have too many things. I suggest you if you find something remaining (that I could forgot) to open a specific issue (and use an adapted name to the issue!).

**Comment by Nilvus** [comment]
Oh, and before posting again, think about update master and test again before. We are already more than 40 commits ahead of your last build. And some things have been improved and fixed in culing/preview modes for example. As well as in metadata things.

**Comment by RawConvert** [comment]
@Nilvus , I've rebuilt today but selection by clicking is still not working and I very respectfully suggest this is worth a look by the devs.

First though, your comment above about clicking selected/unselected and this being normal behaviour.  I think we must be at cross-purposes.  What I meant was that if I click an unselected image, then it should become selected and the border should change.  Bit it doesn't.  And if I click a selected image, it should become unselected, and the border change back.  But it doesn't.  This is the issue I've been having with Master for a week-ish.

Various other selection mechanisms seem fine.  Clicking "select all" works - all images light up.  Similarly "select none" switches all off.  Hovering over an image causes it to light up (and I no longer seem to have the odd intermittent behaviour I mentioned above).  Double click invokes darktable.

Is it my system?  I don't know for sure, but it behaves fine apart from new dt builds.  The dt 3.4 release seems fine and does not have the click-select problem.

Whilst I'm out of my depth with the tec side of things, I did look at what GTK was being used, as I keep the terminal logs.  Both 3.4  and Master today quote GTK 3.22.30.

Before building I renamed the folder "darktable" in /opt to something else, so hopefully this cannot be causing any problems. In any case I think my build process doesn't use this location.
Also I built into a new folder, so there was no pre-existing build folder to cause problems.

I run dt like this -
`$HOME/DT-Mast-23Jan21/bin/darktable --cachedir $HOME/.cache/darktable-23jan --configdir $HOME/.config/darktable-23Jan`
Does dt keep state info or settings/preferences anywhere except the cache and config folders?  Is there perhaps something at another location which is corrupt and causing the problem?

So in summary I'm rather stuck!  Any ideas much appreciated.

**Comment by Nilvus** [comment]
Indeed, your behavior described in last comment is not normal. I can't reproduce and I'm asking if that can be related to your Gtk version. I don't have access to my computer to check which one I have but quite different.

As this issue was related to not only that and title not clear about the issue, I suggest you to open a new issue more precise about your issue (your last comment seems really clear) and ping @AlicVB as he works on lighttable. Link also that issue, as a reminder. Thanks.

**Comment by AlicVB** [comment]
@RawConvert : yes, what you describe about lighttable is definitely an issue. This need investigations, as I can't reproduce (of course, otherwise I would have fix it ;)
As @Nilvus as said maybe the best way is to open a new issue and ping me (with @AlicVB ) in the comment. Thanks !
